### PR TITLE
fmtlog: set fmt version to 10

### PR DESCRIPTION
**Pull request by Chi-EEE**
https://github.com/xmake-io/xmake-repo/issues/4806

### CODE REVIEW

File: packages/f/fmtlog/xmake.lua [hunk 1]
```diff
@@ -8,7 +8,7 @@ package("fmtlog")
     add_versions("v2.2.1", "9bc2f1ea37eece0f4807689962b529d2d4fa07654baef184f051319b4eac9304")
     add_versions("v2.1.2", "d286184e04c3c3286417873dd2feac524c53babc6cd60f10179aa5b10416ead7")
 
-    add_deps("cmake", "fmt")
+    add_deps("cmake", "fmt 10.2.1")
```
Comment (SirLynix) [inline packages/f/fmtlog/xmake.lua:11]: isn't `fmt 10` enough? this allows for minor and patch version mismatch

Comment (Chi-EEE) [reply packages/f/fmtlog/xmake.lua:11]: yes that works, thanks

File: packages/f/fmtlog/xmake.lua [hunk 2]
```diff
@@ -8,7 +8,7 @@ package("fmtlog")
     add_versions("v2.2.1", "9bc2f1ea37eece0f4807689962b529d2d4fa07654baef184f051319b4eac9304")
     add_versions("v2.1.2", "d286184e04c3c3286417873dd2feac524c53babc6cd60f10179aa5b10416ead7")
 
-    add_deps("cmake", "fmt")
+    add_deps("cmake", "fmt 10")
```
Comment (waruqi) [inline packages/f/fmtlog/xmake.lua:11]: why not `>=10`?

Comment (SirLynix) [reply packages/f/fmtlog/xmake.lua]: this would allow for fmt 11 which won't compile here, I'm making a better fix

Comment (star-hengxing) [reply packages/f/fmtlog/xmake.lua:11]: I think `fmt <11` is better.

Comment (SirLynix) [reply packages/f/fmtlog/xmake.lua]: the issue is that the library uses unstable API from fmt (fmt::detail) and will be updated to support fmt 11 in the future, we need to match the package version with a certain fmt version

**Pull request merged**